File: 2022/python/day-02/part1.py
def calculate_score(filename: str) -> int:
    with open(filename, 'r') as f:
        opponent_moves, player_moves = zip(
            *[line.strip().split(" ") for line in f.readlines()])

    equivalent_moves = {'X': 'A', 'Y': 'B', 'Z': 'C'}
    player_wins = {'Y': 'A', 'Z': 'B', 'X': 'C'}
    move_points = {'X': 1, 'Y': 2, 'Z': 3}

    total_points = 0

    for opponent_move, player_move in zip(opponent_moves, player_moves):
        total_points += move_points[player_move]

        if opponent_move == equivalent_moves[player_move]:
            total_points += 3
        elif opponent_move == player_wins[player_move]:
            total_points += 6

    return total_points


def main():
    solution = calculate_score('./input.txt')
    print(solution)


main()

File: 2022/AoCJava/src/Day2/Part1.java
package Day2;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.Hashtable;

public class Part1 {
    public static void main(String[] args) throws Exception {
        BufferedReader input = new BufferedReader(new FileReader("inputs/day2.txt"));
        
        Hashtable<String, String> equivalentMoves = new Hashtable<>();
        equivalentMoves.put("X", "A");
        equivalentMoves.put("Y", "B");
        equivalentMoves.put("Z", "C");

        Hashtable<String, String> playerWins = new Hashtable<>();
        playerWins.put("Y", "A");
        playerWins.put("Z", "B");
        playerWins.put("X", "C");

        Hashtable<String, String> movePoints = new Hashtable<>();
        movePoints.put("X", "1");
        movePoints.put("Y", "2");
        movePoints.put("Z", "3");

        int totalPoints = 0;

        String line;

        while ((line = input.readLine()) != null) {
            line = line.trim();
            String[] moves = line.split(" ");
            totalPoints += Integer.parseInt(movePoints.get(moves[1]));

            if (moves[0].equals(equivalentMoves.get(moves[1]))) {
                totalPoints += 3;
            } else if (moves[0].equals(playerWins.get(moves[1]))) {
                totalPoints += 6;
            }
        }

        System.out.println(totalPoints);
    }
}
